pkg/task/task.go

```diff
@@ -16,6 +16,8 @@ import (
 type Task interface {
 	Run(ctx context.Context, commandContext *CommandContext) Task
 	Name() string
+	Checkpoint() *CompletedTask
+	Restore(ctx context.Context, commandContext *CommandContext, checkpoint *TaskCheckpoint) (Task, error)
```

Comment:
It might be more flexible if we pass the `CompletedTask` instead of the `TaskCheckpoint`?

Reply:
good point! updated